test/Tester/Forwarding/ShutdownSiloTests.cs

```diff
@@ -43,6 +48,76 @@ public override TestCluster CreateTestCluster()
             await promisesAfterShutdown;
         }
 
+        [Fact, TestCategory("Functional"), TestCategory("Forward")]
+        public async Task SiloGracefulShutdown_NoFailureOnGatewayShutdown()
+        {
+            const int numberOfGrains = 2 * NumberOfSilos;
+
+            var grains = new List<ICounterGrain>();
+            for (var i = 0; i < numberOfGrains; i++)
+            {
+                grains.Add(await GetCounterGrainOnPrimary());
+            }
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            var promise = CallIncrement(grains, cancellationTokenSource.Token);
+
+            await Task.Delay(1000);
+            HostedCluster.StopSilo(HostedCluster.SecondarySilos.First());
+            await Task.Delay(1000);
+
+            cancellationTokenSource.Cancel();
+
+            var numberOfCalls = await promise;
+            foreach (var g in grains)
+            {
+                var counterValue = await g.GetValue();
+                Assert.True(numberOfCalls.Item1 == counterValue, $"numberOfCalls={numberOfCalls.Item1} counterValue={counterValue} exceptionCounter={numberOfCalls.Item2}");
+            }
+        }
+
+        private static async Task<Tuple<int, int>> CallIncrement(IList<ICounterGrain> grains, CancellationToken token)
+        {
+            var counter = 0;
+            var tasks = new List<Task>();
+            while (!token.IsCancellationRequested)
+            {
+                foreach (var g in grains)
+                {
+                    tasks.Add(g.IncrementValue());
+                }
+                counter++;
```

Comment:
`counter == tasks.Count / grains.Count`. But I kept the counter so it is easier to understand